targets/csharp/source/Plugins/CloudScript/source/PlayFabFunctionContexts.cs

```diff
@@ -0,0 +1,245 @@
+﻿#if NETSTANDARD2_0
+namespace PlayFab.Plugins.CloudScript
+{
+    using PlayFab.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Used for extracting the data fields incoming as payload to Azure Functions into their respective objects. 
+    /// Meant to be used with basic HTTP based Azure Functions.
+    /// Provides  fields through ApiSettings and AuthenticationContext, the profile of the caller, as well as the 
+    /// arguments passed to the execution request.
+    /// </summary>
+    /// <typeparam name="TFunctionArgument">Type of FunctionArgument</typeparam>
+    public class FunctionContext<TFunctionArgument>
```

Comment:
I'd like to make sure my earlier comments about TFunctionArgument weren't lost or hidden.
This solution will work for the CSharpSDK, but not Unity.
And two solutions with totally different implementations will have an increased maintenance cost.

Reply:
So this piece of code is meant to work only and purely from Azure Functions. In other words, thinking about Unity when you write an Azure Function in C# .Net Standard 2.0 is not a concern imho. Things like this and the ability to focus on one target environment when only one environment is applicable to a use case is the reason why we decided to isolate this plugin from the rest of the SDK.

Reply:
Tl;dr, think of the PlayFabCloudScriptPlugin as a library you install on your Azure Functions App. As long as it works in the context of an Azure Functions App, it's serving its purpose well.